ze/tracer_ze_helpers.include.c

```diff
@@ -1025,14 +1026,16 @@ static void thapi_sampling_energy() {
       }
     }
     if (tracepoint_enabled(lttng_ust_ze_sampling, computeEngine)){
+      computeEngineData computeE[_sampling_subDeviceCount[i]];
```

Comment:
Array will be empty if you have no sub device. Maybe set `_sampling_subDeviceCount[i]` to1 in this case